src/Evolution/Systems/GeneralizedHarmonic/UpwindPenaltyCorrection.cpp

```diff
@@ -0,0 +1,198 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/UpwindPenaltyCorrection.hpp"
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+namespace GeneralizedHarmonic {
+template <size_t Dim>
+void UpwindPenaltyCorrection<Dim>::package_data(
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+        packaged_v_spacetime_metric,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim, Frame::Inertial>*>
+        packaged_v_zero,
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+        packaged_v_plus,
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+        packaged_v_minus,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim, Frame::Inertial>*>
+        packaged_n_times_v_plus,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim, Frame::Inertial>*>
+        packaged_n_times_v_minus,
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+        packaged_gamma2_v_spacetime_metric,
+    const gsl::not_null<tnsr::a<DataVector, 3, Frame::Inertial>*>
+        packaged_char_speeds,
+
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& v_spacetime_metric,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& v_zero,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& v_plus,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& v_minus,
+    const std::array<DataVector, 4>& char_speeds,
+    const Scalar<DataVector>& constraint_gamma2,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& interface_unit_normal)
+    const noexcept {
+  for (size_t storage_index = 0; storage_index < v_spacetime_metric.size();
+       ++storage_index) {
+    packaged_v_spacetime_metric->operator[](storage_index) =
+        char_speeds[0] * v_spacetime_metric[storage_index];
```

Comment:
I understand why you're multiplying by the char speeds while packaging instead of later, but then this makes the names of the variables confusing. I.e. `packaged_v_spacetime_metric` actually is `v_spacetime_metric * char_speeds[0]`. Maybe consider renaming the variables to things like `packaged_lambda_v_spacetime_metric`, or `packaged_char_speed_v_spacetime_metric` or something along those lines? Or at least a comment in the hpp file that these variables will actually not store what the name implies they will?